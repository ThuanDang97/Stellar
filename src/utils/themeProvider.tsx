import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
