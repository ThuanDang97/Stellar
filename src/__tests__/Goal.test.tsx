import React from 'react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Page
import Goal from '../pages/goal'

describe('Goal', () => {
  it('renders goal page unchanged', () => {
    const { asFragment } = renderWithTheme(<Goal />)

    expect(asFragment()).toMatchSnapshot()
  })
})
