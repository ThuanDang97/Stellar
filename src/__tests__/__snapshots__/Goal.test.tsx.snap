// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal renders goal page unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__WrapperContent-sc-rqe5sa-6 kBewKp gVWyht"
    display="flex"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 knVWgS"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 knVWgS"
        display="flex"
      >
        <span
          class="TextStyled-sc-9xia41-0 leTdwL"
          color="#C0C0C0"
          data-testid="stepper-value"
          font-weight="400"
        >
          Sign Up
        </span>
        <span
          class="TextStyled-sc-9xia41-0 StepperStyled__ArrowStyled-sc-1rhowde-0 gtHdGO jbeCTy"
        />
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 knVWgS"
        display="flex"
      >
        <span
          class="TextStyled-sc-9xia41-0 exqdNj"
          color="#203F59"
          data-testid="stepper-value"
          font-weight="400"
        >
          Set Goals
        </span>
        <span
          class="TextStyled-sc-9xia41-0 StepperStyled__ArrowStyled-sc-1rhowde-0 gtHdGO jbeCTy"
        />
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 knVWgS"
        display="flex"
      >
        <span
          class="TextStyled-sc-9xia41-0 leTdwL"
          color="#C0C0C0"
          data-testid="stepper-value"
          font-weight="400"
        >
          Connect Bank
        </span>
      </div>
    </div>
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 lpivDj WstJK"
      color="#000"
      font-family="AdobeCleanExtraBold"
      font-size="36"
    >
      What are your Stellar goals?
    </h1>
    <span
      class="TextStyled-sc-9xia41-0 fpvcHn"
    >
      Select all that apply.
    </span>
    <div
      class="ViewStyled-sc-rqe5sa-0 daOeHc"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 etuUCN"
        display="flex"
        width="100%"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Improve My 
 Credit Score
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Build 
 Wealth
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Buy A Car
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Get A Job
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Buy 
 A Home
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Rent 
 A Home
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Get A 
 Credit Card
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Get Better 
 Insurance Rate
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Get Better 
 Interest Rates
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Pay May Bills 
 On Time
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Get A Loan
          </p>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
          cursor="pointer"
          data-testid="card-goal"
          display="flex"
          height="120"
          width="160"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Card Goal"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="TextStyled-sc-9xia41-0 eTpfgb"
            color="#203F59"
            display="-webkit-box"
            font-size="14"
            overflow="hidden"
          >
            Start 
 A Business
          </p>
        </div>
      </div>
      <button
        class="ButtonStyled-sc-1ys9v52-0 ieXaIH"
        data-testid="button"
        disabled=""
        letter-spacing="0.69"
        type="submit"
      >
        Continue
      </button>
    </div>
  </div>
</DocumentFragment>
`;
